src/convertToObject.js

```diff
@@ -5,8 +5,25 @@
  *
  * @return {object}
  */
-function convertToObject(sourceString) {
-  // write your code here
+function convertToObject(str) {
+  const result = {};
+  const listStyle = str.split(';');
+
+  for (const listStyleElement of listStyle) {
+    const styleItem = listStyleElement.trim();
+
+    if (styleItem) {
+      const [property, ...itemParts] = styleItem.split(':');
+
+      if (property && itemParts.length > 0) {
+        const item = itemParts.join(':').trim();
+
+        result[property.trim()] = item;
+      }
+    }
```

Comment:
The use of a for...of loop here contradicts the checklist's preference for iteration methods. Consider using 'split', 'filter', and 'reduce' methods to transform the input string into the desired object.